src/ui/LiveButton.js

```diff
@@ -10,50 +11,79 @@ import {
   UIText,
 } from '@apollosproject/ui-kit';
 import { AnalyticsConsumer } from '@apollosproject/ui-analytics';
+import { PLAY_VIDEO } from '@apollosproject/ui-media-player';
 import {
-  RockAuthedWebBrowser,
   LiveConsumer,
+  RockAuthedWebBrowser,
 } from '@apollosproject/ui-connected';
 
 const LiveCard = styled(({ theme }) => ({
   backgroundColor: theme.colors.lightSecondary,
 }))(Card);
 
+const renderLiveButton = ({ action }) => (
+  <TouchableScale onPress={action}>
+    <LiveCard>
+      <CardContent>
+        <ChannelLabel
+          icon="video"
+          label={
+            <UIText>
+              <UIText bold>{`We're live.`} </UIText>
+              Watch now!
+            </UIText>
+          }
+        />
+      </CardContent>
+    </LiveCard>
+  </TouchableScale>
+);
+
+renderLiveButton.propTypes = {
+  action: PropTypes.func,
+};
+
+const renderMedia = (liveStream, track) => (
+  <Mutation mutation={PLAY_VIDEO}>
+    {(play) =>
+      renderLiveButton({
+        action: () => {
+          play({
+            variables: {
+              mediaSource: liveStream.media.sources[0],
+              isVideo: true,
+            },
+          });
+          track({ eventName: 'Clicked Live Bar' });
+        },
+      })
+    }
+  </Mutation>
+);
+
+const renderWebView = (liveStream, track) => (
+  <RockAuthedWebBrowser>
+    {(openUrl) =>
+      renderLiveButton({
+        action: () => {
+          openUrl(liveStream.webViewUrl);
+          track({ eventName: 'Clicked Live Bar' });
+        },
+      })
+    }
+  </RockAuthedWebBrowser>
+);
+
 const LiveNowButton = ({ contentId }) => (
   <LiveConsumer contentId={contentId}>
     {(liveStream) =>
       liveStream ? (
         <AnalyticsConsumer>
-          {({ track }) => (
-            <RockAuthedWebBrowser>
-              {(openUrl) => (
-                <TouchableScale
-                  onPress={() => {
-                    openUrl(
-                      'https://live.newspring.cc/',
-                      {},
-                      { useRockToken: true }
-                    );
-                    track({ eventName: 'Clicked Live Bar' });
-                  }}
-                >
-                  <LiveCard>
-                    <CardContent>
-                      <ChannelLabel
-                        icon="video"
-                        label={
-                          <UIText>
-                            <UIText bold>{`We're live.`} </UIText>
-                            Watch now!
-                          </UIText>
-                        }
-                      />
-                    </CardContent>
-                  </LiveCard>
-                </TouchableScale>
-              )}
-            </RockAuthedWebBrowser>
-          )}
+          {({ track }) =>
+            liveStream && get(liveStream, 'media.sources[0].uri')
```

Comment:
you don't need this `liveStream` check here, you already have that on line 80.

Reply:
👍 That's a great catch!